modules/transaction-payment/src/lib.rs

```diff
@@ -825,31 +821,101 @@ where
 					<T as Config>::Currency::ensure_can_withdraw(who, fee, reason, new_free_balance).is_ok()
 				});
 		if native_is_enough {
-			return;
+			None
+		} else {
+			Some(fee.saturating_add(native_existential_deposit.saturating_sub(total_native)))
```

Comment:
```suggestion
			Some(fee.saturating_add(native_existential_deposit))
```

Otherwise I think it will have some issue like when user have a lot of locked native token. Then the amount will become zero.

Reply:
current result is fee+(ed-total), even though total(reserve balance) is too large, ed-total=0 thanks to saturating_sub, so the result amount would't be zero instead of fee.  if change to (fee+ed)-total then it's possible getting 0 if total have large reserve balance.